### PR TITLE
cleaning newNodeController from unsupported kubelet version

### PR DESCRIPTION
**Pull request by yastij**
**What this PR does / why we need it**: this part of the #52356 effort

**Which issue this PR fixes** : fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by cblecker** [comment]
Looks like there is dependencies on `forcefullyDeletePod`
```
W1005 19:04:05.945] # k8s.io/kubernetes/pkg/controller/node
W1005 19:04:05.946] pkg/controller/node/node_controller.go:1237: nc.forcefullyDeletePod undefined (type *Controller has no field or method forcefullyDeletePod)
W1005 19:04:05.946] pkg/controller/node/node_controller.go:1241: nc.forcefullyDeletePod undefined (type *Controller has no field or method forcefullyDeletePod)
```

**Comment by yastij** [comment]
@cblecker - yup I know, I still have to add its deletion here from #52402

**Comment by yastij** [comment]
@gmarek - ping for lgtm/approve

File: pkg/controller/node/node_controller.go [hunk 1]
```diff
@@ -285,9 +284,6 @@ func NewNodeController(
 		serviceCIDR:            serviceCIDR,
 		allocateNodeCIDRs:      allocateNodeCIDRs,
 		allocatorType:          allocatorType,
-		forcefullyDeletePod: func(p *v1.Pod) error {
-			return util.ForcefullyDeletePod(kubeClient, p)
```
Comment (gmarek) [inline pkg/controller/node/node_controller.go]: You should also delete function definition. I think it's used only here.

Comment (yastij) [reply pkg/controller/node/node_controller.go]: indeed missed this one

**Comment by yastij** [comment]
@gmarek - updated

**Pull request merged**